Gossipsub v1.1 Blog post

daviddias opened the pull request:
Just missing the decision to include the mockchain example and a nice gif for the baseline test plan. Other than that, good to review :)

daviddias commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+We are also releasing a brand new libp2p Test Plan that uses Testground to benchmark Gossipsub v1.1.
+
+<p align="center">
+  <img width="60%" src="/header_images/094-gossipsub-v1.1-headerimage.jpeg">
+</p>
+
+TODO INSERT IMAGE HERE of the JUPYTER NOTEBOOK
```
@yusefnapora do you happen to have a nice screenshot (gift would be 1000x better) of what looks like to run the baseline test plan ? https://github.com/libp2p/test-plans/tree/master/pubsub

daviddias commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+You can find the code for this example at https://github.com/libp2p/go-libp2p-examples/tree/master/pubsub/chat
+
+### Mockchain example
+
+TODO write this
```
@yusefnapora, were you able to make progress on this or not quite? Should we remove it from this launch?

yusefnapora replied:
I don't think I'll be able to finish it before launch 😞

daviddias replied:
All good. I've removed it :)

momack2 commented:
Be careful to use the right aspect ratio on the header. It needs to be 500px by 250px or equivalent or the grid breaks: 
![image](https://user-images.githubusercontent.com/618519/82306399-de6a0400-9973-11ea-874e-6f6e1f969c19.png)

daviddias commented:
I see! Thanks for catching that. Fixing

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,80 @@
+
+1) [Peer scoring](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#peer-scoring): peers now monitor their directly connected neighbours using a score function that reflects the peer’s usefulness. Well-behaving peers are kept in the mesh, while badly-behaving ones are dropped in order to protect the network from malicious actors.
+
+2) [Adaptive gossip dissemination](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#adaptive-gossip-dissemination): In this new version, you can adjust the amount of peers in which your node gossips to with the gossip factor. This enables you to increase/decrease the amount of gossip, while keeping a baseline.
```
```suggestion
2) [Adaptive gossip dissemination](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#adaptive-gossip-dissemination): In this new version, you can adjust the `gossip factor` to control the number of peers your node gossips with. This enables you to increase/decrease the amount of gossip, while keeping a baseline.
```
what do you mean by "amount of gossip"? That isn't clear to me? Number of nodes I peer with? number of intermediate hops my message will propogate?

daviddias replied:
In Gossipsub v1.0, you always sent Gossip to a fixed amount of peers, in v1.1, you can adjust that with a factor, increasing/decreasing the number of peers that you send the gossipsub to.

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,80 @@
+
+2) [Adaptive gossip dissemination](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#adaptive-gossip-dissemination): In this new version, you can adjust the amount of peers in which your node gossips to with the gossip factor. This enables you to increase/decrease the amount of gossip, while keeping a baseline.
+
+3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting): A new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically grafted to improve the mesh’s health in terms of median score of participating peers. 
```
```suggestion
3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting): Grafting is a new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically inserted into the mesh to improve the median score of participating peers. 
```
Shouldn't use "grafting" to explain what grafting is. Assume it's inserting them into the mesh? maybe "connected" is a better word?

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,80 @@
+
+3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting): A new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically grafted to improve the mesh’s health in terms of median score of participating peers. 
+
+4) [Prune Peer eXchange](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#prune-backoff-and-peer-exchange): with Prune Peer exchange a node receives a set of recommended peers to connect when it gets dropped from some other peer’s mesh. With this, we’ve reduced the dependency on ambient peer discovery mechanisms.
```
```suggestion
4) [Prune Peer eXchange](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#prune-backoff-and-peer-exchange): Prune Peer exchange gives a node a set of recommended peers to connect when it gets dropped from another peer’s mesh, helping reduce the dependency on ambient peer discovery mechanisms.
```

daviddias commented:
Fixed the header image and adjusted the title to just take 1 line to further match the grid
<img width="1336" alt="image" src="https://user-images.githubusercontent.com/1211152/82307344-23178e80-99b8-11ea-8626-9d6f1ea40bec.png">

vyzo commented:
Can we add something for the outbound connection quotas as well? It's just in, and it's a nice security feature.

vyzo commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+**2) [Adaptive gossip dissemination](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#adaptive-gossip-dissemination)**: In this new version, you can adjust the `gossip factor` to control the number of peers your node gossips with. This enables you to increase/decrease the amount of gossip, while keeping a baseline.
+
+**3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting)**: Grafting is a new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically inserted into the mesh to improve the median score of participating peers. 
```
```suggestion
**3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting)**: Opportunistic Grafting is a new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically inserted into the mesh to improve the median score of participating peers. 
```

daviddias replied:
```suggestion
**3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting)**: A new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically inserted into the mesh to improve the median score of participating peers. 
```

vyzo commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+**3) [Opportunistic Grafting](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#opportunistic-grafting)**: Grafting is a new mechanism used to cherry pick well-behaving peers that are currently not part of a node’s mesh. Those well-behaving peers are then opportunistically inserted into the mesh to improve the median score of participating peers. 
+
+**4) [Prune Peer eXchange](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#prune-backoff-and-peer-exchange)**: Prune Peer exchange gives a node a set of recommended peers to connect when it gets dropped from another peer’s mesh, helping reduce the dependency on ambient peer discovery mechanisms.
```
we also have to mention the backoff I think

daviddias replied:
Folks can always read the spec. For this blog post, our goal is to have something that works as a highlights/gist, so that folks can feel they grok it but also know where to look for more details (i.e. the spec) when they decide to.

daviddias commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+We are also releasing a brand new libp2p Test Plan that uses [Testground](https://github.com/testground/testground) to benchmark Gossipsub v1.1.
+
+<p align="center">
+  <img width="60%" src="/img/gossipsubv1.1-jupyter.jpeg">
+</p>
```
@yusefnapora is preparing a gif to replace this illustration.

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+---
+date: 2020-05-19
+url: 2020-05-19-gossipsub-v1.1
```
```suggestion
date: 2020-05-20
url: 2020-05-20-gossipsub-v1.1
```

cwaring commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+---
+
+<p align="center">
+  <img width="60%" src="/header_images/094-gossipsub-v1.1-headerimage+grid.jpeg">
```
```suggestion
  <img src="/header_images/094-gossipsub-v1.1-headerimage+grid.jpeg" style="max-width:600px">
```

daviddias commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+We’ve prepared a brand new example for you to give the latest Gossipsub a try. In this example, you get to experiment with a chat application powered by libp2p. It will enable you to join different channels by subscribing to PubSub topics and seeing other peers joining the room as they subscribe to the topic.
+
+<p align="center">
+  <img width="60%" src="https://raw.githubusercontent.com/libp2p/go-libp2p-examples/master/pubsub/chat/chat-example.gif">
```
```suggestion
  <img src="https://raw.githubusercontent.com/libp2p/go-libp2p-examples/master/pubsub/chat/chat-example.gif" style="max-width:600px">
```

daviddias commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+We are also releasing a brand new libp2p Test Plan that uses [Testground](https://github.com/testground/testground) to benchmark Gossipsub v1.1.
+
+<p align="center">
+  <img width="60%" src="/img/gossipsubv1.1-jupyter.gif">
```
```suggestion
  <img src="/img/gossipsubv1.1-jupyter.gif" style="max-width:600px">
```

cwaring replied:
600px might not be the right size for all of these but play around and see what looks best 👀

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+### Chat application example
+
+We’ve prepared a brand new example for you to give the latest Gossipsub a try. In this example, you get to experiment with a chat application powered by libp2p. It will enable you to join different channels by subscribing to PubSub topics and seeing other peers joining the room as they subscribe to the topic.
```
```suggestion
We’ve prepared a brand new example for you to give the latest Gossipsub a try. In this example, you get to experiment with a chat application powered by libp2p. It will enable you to join different channels by subscribing to PubSub topics and see other peers joining the room as they subscribe to the topic.
```

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+We’ve prepared a brand new example for you to give the latest Gossipsub a try. In this example, you get to experiment with a chat application powered by libp2p. It will enable you to join different channels by subscribing to PubSub topics and seeing other peers joining the room as they subscribe to the topic.
+
+<p align="center">
+  <img src="https://raw.githubusercontent.com/libp2p/go-libp2p-examples/master/pubsub/chat/chat-example.gif" style="max-width:640px">
```
```suggestion
  <a href="https://github.com/libp2p/go-libp2p-examples/tree/master/pubsub/chat"><img src="https://raw.githubusercontent.com/libp2p/go-libp2p-examples/master/pubsub/chat/chat-example.gif" style="width:100%"></a>
```
This is currently too small to be readable. Trying to make this larger.

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+We are also releasing a brand new libp2p Test Plan that uses [Testground](https://github.com/testground/testground) to benchmark Gossipsub v1.1.
+
+<p align="center">
+  <img src="/img/gossipsubv1.1-jupyter.gif" style="max-width:640px">
```
```suggestion
  <a href="https://github.com/libp2p/test-plans/tree/master/pubsub"><img src="/img/gossipsubv1.1-jupyter.gif" style="width:100%"></a>
```

momack2 commented on content/post/094-gossipsub-v1.1.md:
```diff
@@ -0,0 +1,79 @@
+
+That’s it for now. Hope you enjoy the release and happy gossiping!
+
+The Gossipsub Task Force
+
+David Dias, Dmitris Vyzovitis, Yiannis Psaras, Yusef Napora, Dirk McCormick
```
```suggestion
**The Gossipsub Task Force** - David Dias, Dmitris Vyzovitis, Yiannis Psaras, Yusef Napora, Dirk McCormick
```
